src/briefcase/platforms/android/apk.py

```diff
@@ -1 +1,242 @@
+            / app.formal_name
+            / "app"
+            / "build"
+            / "outputs"
+            / "apk"
+            / "debug"
+            / "app-debug.apk"
+        )
+
+    def distribution_path(self, app):
+        return self.binary_path(app)
+
+    def verify_tools(self):
+        """
+        Verify that we can download a support package for this Python version,
+        then ensure the Android development tools are installed.
+        """
+        if self.python_version_tag != "3.7":
+            raise BriefcaseCommandError("""\
+Found Python version {self.python_version_tag}. Android packaging currently
+requires Python 3.7.""".format(self=self))
+
+        if not self.android_sdk_path.exists():
+            print("Setting up Android SDK...")
+            try:
+                android_sdk_zip_path = self.download_url(
+                    url=self.android_sdk_url,
+                    download_path=Path.home() / ".briefcase" / "tools",
+                )
+            except requests_exceptions.ConnectionError:
+                raise NetworkFailure("downloading Android SDK")
+            with ZipFile(android_sdk_zip_path) as android_sdk_zip:
+                android_sdk_zip.extractall(path=self.android_sdk_path)
+            # Remove the ZIP file; it has no purpose now that it is extracted.
+            android_sdk_zip_path.unlink()
+            # Set executable permissions. The ZipFile module does not extract
+            # these permissions, but Linux & macOS need them.
+            # TODO: Test this on Windows.
+            tools_bin = self.android_sdk_path_tmp / "tools" / "bin"
+            for binpath in tools_bin.glob('*'):
+                binpath.chmod(0o755)
+
+            print("Ensuring all Android SDK licenses are accepted...")
+            self.subprocess.run(
+                [tools_bin / "sdkmanager", "--licenses"],
+                check=True,
+                cwd=self.android_sdk_path,
+            )
+
+
+# The way to split this up is that you might start with more
+# stuff externally specified.
+
+
+class ApkCreateCommand(ApkMixin, CreateCommand):
+    description = "Create and populate an Android APK."
+
+
+class ApkUpdateCommand(ApkMixin, UpdateCommand):
+    description = "Update an existing Android APK."
+
+
+class ApkBuildCommand(ApkMixin, BuildCommand):
+    description = "Build an Android APK."
+
+    @property
+    def android_sdk_url(self):
+        # TODO: Add test validating that, if this is mocked out for a sentinel
+        # ZIP file, we surely unpack it into android_sdk_path.
+        """The Android SDK URL appropriate to this operating system."""
+        # The URLs described by the pattern below have existed since
+        # approximately 2017, and the code they download has a built-in
+        # updater. I hope they will work for many years.
+        return "https://dl.google.com/android/repository/" + (
+            "sdk-tools-{os}-4333796.zip".format(os=self.host_os.lower()))
```

Comment:
Seems odd that it's here, and not in the mixin. If this is defined on Build, what happens when verify_tools() is involved on a standalone Create/Update?

Reply:
Good question. I'll investigate.

Reply:
Fixed in a follow-up, pushed to this pull request.